plugins/PostProcessingPlugin/scripts/X40Snapshot.py

```diff
@@ -0,0 +1,80 @@
+from UM.Logger import Logger
```

Comment:
We'd like to request that this script gets merged with the existing post-processing script. And I can help out a bit with the transfer if necessary; just ask :)

This is how I envision it working nicely for others and for the Weedo users:
* In CreateThumbnail.py, add a new parameter to specify the format, a drop-down with two options, "Octoprint" and yours. Octoprint should be the default. Your option could be called "Weedo" since your W220 etc. commands seem to be specific to Weedo printers. Or maybe you could call it "W221".
* In the `_encodeSnapshot` function you can read this parameter and choose between PNG+base64 for Octoprint and JPG+Hex for Weedo.
* The `_convertSnapshotToGcode` needs to then choose different prefixes, etc. for Weedo but can otherwise stay mostly the same.

This solution is desirable because of two sides:
* It reduces code duplication.
* It reduces user confusion as to which post-processing script to choose.

In early versions of Cura we had various flavours of Pause at Height for different g-code flavours and it was causing people to be confused as to which one they should choose. Each script had different parameters and different features. Combining them made things more clear for users and also allowed us to give all of these pause methods the same features, resolve bugs all at once instead of needing to test all scripts separately, etc.

Reply:
This script has been removed completely now. No support for these screenshots then, but it should work without too.